packages/axelar-wasm-std/src/msg_id/tx_hash_event_index.rs

```diff
@@ -0,0 +1,199 @@
+use core::fmt;
+use std::{fmt::Display, str::FromStr};
+
+use cosmwasm_std::HexBinary;
+use regex::Regex;
+
+use super::Error;
+use crate::{hash::Hash, nonempty};
+pub struct HexTxHashAndEventIndex {
+    pub tx_hash: Hash,
+    pub event_index: u32,
+}
+
+impl HexTxHashAndEventIndex {
+    pub fn tx_hash_as_hex(&self) -> nonempty::String {
+        format!("0x{}", HexBinary::from(self.tx_hash).to_hex())
+            .try_into()
+            .expect("failed to convert tx hash to non-empty string")
+    }
+}
+
+impl FromStr for HexTxHashAndEventIndex {
+    type Err = Error;
+    fn from_str(message_id: &str) -> Result<Self, Self::Err>
+    where
+        Self: Sized,
+    {
+        const PATTERN: &str = "^(0x[0-9a-f]{64})-(0|[1-9][0-9]*)$";
```

Comment:
I agree regex is probably not the most performant solution, but I think it's probably cleaner and safer compared to manually checking everything. Also, if we go with regex and then discover there is a gas issue, we can always change the implementation, since it would not break the protocol. @cgorenflo thoughts?

Reply:
agreed. Let the people who created the regex library worry about implementing the validation. This way it's much easier to ensure no edge cases have been missed